docs/source/peer_ledger_snapshot.md

```diff
@@ -0,0 +1,164 @@
+```
+
+For example:
+
+```
+peer snapshot submitrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+**If you give a ledger height of `0`, the snapshot is taken immediately. This is useful for cases when an organization is interested in generating a snapshot that will be used by one of its own peers and does not intend to share the data with another organization. Do not take these "immediate" snapshots in cases when snapshots will be evaluated from multiple peers, as it increases the likelihood that the snapshots will be taken at different ledger heights.**
+
+If the request is successful, you will see a `Snapshot request submitted successfully` message.
+
+You can list the pending snapshots by issuing a command similar to:
+
+```
+peer snapshot listpending -C testchannel --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+You will see a response similar to:
+
+```
+Successfully got pending snapshot requests [1000]
+```
+
+When a snapshot has been generated for a particular block height, the pending request for that block height will no longer appear in the list. You can also verify that a snapshot has been create successfully by looking at the peer logs or by checking the expected location of the snapshot, `{snapshotsFolder}/completed/{channelName}/{lastBlockNumberInSnapshot}`, for the snapshot.
+
+To delete a snapshot request, simply exchange `submitrequest` with `cancelrequest`. For example:
+
+```
+peer snapshot cancelrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+If you submit the `listpending` command again, the snapshot should no longer appear.
+
+### Contents of a snapshot
+
+Snapshots are stored on a peer in a snapshot folder that is further organized by channel name and the block number when the snapshot was taken: `{snapshotsFolder}/completed/{channelName}/{lastBlockNumberInSnapshot}`. Once the peer generates a snapshot in this folder, the peer does not use that folder for any purpose and it is safe to transfer or compress the snapshot using external tools.
+
+As mentioned above, the snapshot contains a folder with files for the different data items listed below:
+
+* **Public state**
+  * This includes the latest state of all of the assets on the channel (for example, the current owner of a car).
+* **Private data hashes**
+  * Recall from our documentation on private data that while the actual transaction data of a private data transaction is not stored on the public ledger of the channel, hashes of the data are stored. This allows the private data to be verified against the hash, for example by an organization that is added to a private data collection. These hashes are included in the snapshot so that new organizations can verify them against the private data they receive.
```

Comment:
>These hashes are included in the snapshot so that new organizations can verify them against the private data they receive

I noticed this addition in the last commit. I would like to refrain from this as there is more to the hashes than this. In fact, hashes are present with every peer whether the peer is eligible for the private data or not.

Reply:
Well, I think we do need to say something here about what these hashes are for, even if it's not the whole truth. Maybe you can suggest an alternative here?

Reply:
I had suggested a sentence in a [previous comment](https://github.com/hyperledger/fabric/pull/2043#discussion_r512322065). We have already mentioned before that the world state is one of the data items in the snapshot.